src/core_atmosphere/mpas_atm_core.F

```diff
@@ -862,15 +862,55 @@ subroutine atm_do_timestep(domain, dt, itimestep)
       xtime_s = (s + s_n / s_d)
 
 
+      !
+      ! This is a call to all of the slow, column-oriented physics, except for resolved
+      ! cloud microphysical processes. The microphysics is handled after the dynamics so 
+      ! that any supersaturation caused by various dynamical processes (advection, etc) 
+      ! may be removed before the end of each time step.
+      !
 #ifdef DO_PHYSICS
       !proceed with physics if moist_physics is set to true:
       if(moist_physics) then
          call physics_timetracker(domain,dt,clock,itimestep,xtime_s)
-         call physics_driver(domain,itimestep,xtime_s)
+         call physics_part1(domain,itimestep,xtime_s)
+      endif
+#endif
+
+      call dynamics_timestep(domain, dt, currTime, itimestep)
+
+      !
+      ! This part is the microphysics, which needs to happen after the
+      ! dynamics time stepping.
+      !
+#ifdef DO_PHYSICS
+      !proceed with physics if moist_physics is set to true:
+      if(moist_physics) then
+         call physics_part2(domain,itimestep,xtime_s)
       endif
 #endif
 
-      call atm_timestep(domain, dt, currTime, itimestep)
+      !
+      ! If this is a regional domain, adjustments are required for the
+      ! lateral boundary conditions due to the microphysics.
+      !
+!DAVE if this really is due to MP, the moist physics switch is legit??
+#ifdef DO_PHYSICS
```

Comment:
There are scalars in the LBC update, but there are other variables also. Do we really not do LBC update if no moist scalars?

Reply:
Also, is the ifdef for DO_PHYSICS correct for doing LBC update? Could there be a regional domain with no moist fields on the lateral boundaries?